custom/icds_reports/reports/governance_apis.py

```diff
@@ -41,6 +52,40 @@ def base_data(row_data):
     return data_rows, data.count()
 
 
+@icds_quickcache(['length', 'year', 'month', 'order', 'query_filters'], timeout=30 * 60)
+def get_vhnd_data(length, year, month, order, query_filters):
+    data = GovVHNDView.objects.filter(
+        month=date(year, month, 1),
+        **query_filters
+    ).order_by(*order).values(
+        'state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name', 'month',
+        'vhsnd_date_past_month', 'anm_mpw_present', 'asha_present', 'child_immu', 'anc_today', 'awc_id'
+    )
+
+    paginated_data = data[:length]
+
+    def base_data(row_data):
+        return dict(
+            state=get_value_or_data_not_entered(row_data, 'state_name'),
```

Comment:
I think this can be simplified like this https://github.com/dimagi/commcare-hq/pull/26325/files#diff-2c73ea571072adbc6bb48114d0b66e7dR48. It will be much less code and more readable

Reply:
there are some changes to the return names and value formats. So added this way

Reply:
YOu could use the F function as in the above link

Reply:
done